godtools/App/Features/Tools/SwiftUI/Favorites/AllFavoriteToolsView/AllFavoriteToolsViewModel.swift

```diff
@@ -84,5 +87,50 @@ extension AllFavoriteToolsViewModel: ToolCardViewModelDelegate {
     
     func openToolButtonTapped(resource: ResourceModel) {
         flowDelegate?.navigate(step: .toolTappedFromFavoritedTools(resource: resource))
+        trackOpenFavoritedToolButtonAnalytics()
     }
 }
+
+// MARK: - Analytics
+
+extension AllFavoriteToolsViewModel {
+    var analyticsScreenName: String {
+        return "All Favorites"
+    }
+    
+    private var analyticsSiteSection: String {
+        return "home"
+    }
+    
+    private var analyticsSiteSubSection: String {
+        return ""
+    }
+        
+    func pageViewed() {
+        
+        analytics.pageViewedAnalytics.trackPageView(trackScreen: TrackScreenModel(screenName: analyticsScreenName, siteSection: analyticsSiteSection, siteSubSection: analyticsSiteSubSection))
+    }
+    
+    func trackFavoritedToolTappedAnalytics() {
```

Comment:
And can these tracking methods be private as well?

Reply:
Yep good call 🤦‍♀️